src/models/tiles/tile-model.ts

```diff
@@ -17,6 +17,7 @@ export interface ITilePosition {
   tileId: string;
 }
 export interface IDragTileItem extends ITilePosition {
+  newTileId?: string;
```

Comment:
Note that `IDropTileItem` (defined below) differs only from `IDragTileItem` in having a `newTileId` property. Worth considering at some point whether we still need both (`newTileId` is optional in one and required in the other) or some other simplification is possible. There may still be value in having it required in some contexts, but just noting that this adds a bit of redundancy.